### PR TITLE
fix: Run onStart only once in EventViewModel

### LINKED ISSUE #589: Settings Fragment closes on Screen rotation
**Is it a ?**

- [x] Bug Report

- [ ] Feature Request
    
- [ ] Chore
    

**Actual Behaviour**

When user is in Settings Fragment and rotate the screen, the settings fragment is destroyed and Events layout page is loaded.

**Expected Behaviour**

Settings fragment should withstand app rotation.

**Steps to reproduce it**

LogIn -> Select settings option from drawer -> Rotate the screen -> Event's page gets loaded.
ezgif-4-ab1443eee8

**LogCat for the issue**
```

OkHttp: --> GET https://open-event-api.herokuapp.com/v1/users/327/events?page[size]=0 http/1.1
D/OkHttp: --> END GET
D/OkHttp: <-- 200 OK https://open-event-api.herokuapp.com/v1/users/327/events?page[size]=0 (664ms)
D/OkHttp: Connection: keep-alive
D/OkHttp: Content-Type: application/vnd.api+json
D/OkHttp: Etag: 46d23668535421f2fa7b891b628453eb264ca55f
D/OkHttp: Server: gunicorn/19.7.1
D/OkHttp: Date: Mon, 15 Jan 2018 05:07:39 GMT
D/OkHttp: Access-Control-Allow-Origin: *
D/OkHttp: Via: 1.1 vegur
D/OkHttp: {"meta": {"count": 0}, "data": [], "jsonapi": {"version": "1.0"}, "links": {"self": "/v1/events?page%5Bsize%5D=0"}}
D/OkHttp: <-- END HTTP (115-byte body)
I/DatabaseRepository: Saved items of type class org.fossasia.openevent.app.common.data.models.Event in database
I/Logger: List of items successfully loaded with count: 0
I/art: Starting a blocking GC Explicit
I/art: Explicit concurrent mark sweep GC freed 8173(497KB) AllocSpace objects, 0(0B) LOS objects, 40% free, 17MB/29MB, paused 660us total 61.718ms

```
**Screenshots of the issue**

Where-ever possible attach a screenshot of the issue.

**Android version and Phone Model**

Moto G4 Plus
    
**Would you like to work on the issue?**

No

### PR DESCRIPTION
**Pull request by srv-twry**
Fixes #589 

Checklist:

- [x] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Changes: Run onStart only once in the EventViewModel and as a result stop dashboard from overriding the fragments.

Screenshots for the change: 
videotogif_2018 07 02_10 04 00

### CODE REVIEW

**Comment by iamareebjamal** [comment]
But why is it called twice

**Comment by srv-twry** [comment]
Because I am calling it from the `onStart` of the Fragment.

**Comment by iamareebjamal** [comment]
Try onCreateView

**Comment by srv-twry** [comment]
`AppCompatActivity` doesn't have an `onCreateView` method. When I use it, then the app crashes.

First few lines of the log
```
07-02 14:49:47.225 9027-9027/org.fossasia.eventyay E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.fossasia.eventyay, PID: 9027
    android.view.InflateException: Binary XML file line #17: Error inflating class android.support.design.internal.NavigationMenuItemView
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:763)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:482)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:414)
        at android.support.design.internal.NavigationMenuPresenter$NormalViewHolder.<init>(NavigationMenuPresenter.java:302)
        at android.support.design.internal.NavigationMenuPresenter$NavigationMenuAdapter.onCreateViewHolder(NavigationMenuPresenter.java:402)
        at android.support.design.internal.NavigationMenuPresenter$NavigationMenuAdapter.onCreateViewHolder(NavigationMenuPresenter.java:352)
        at android.support.v7.widget.RecyclerView$Adapter.createViewHolder(RecyclerView.java:6685)
        at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5869)
        at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5752)
        at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5748)
        at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2232)

```

**Comment by iamareebjamal** [comment]
Then onCreate should do it

**Comment by srv-twry** [comment]
> Then onCreate should do it

`onCreate` is called everytime the activity is created(after rotation also). The requirement is that the `onStart` function(defined by me) in the ViewModel should be only called once. Two ways to do it is:
1. call it from the constructor of the ViewModel but it is not the recommended approach.
2. Have a flag as I am doing. 
In almost all the methods of the ViewModel we check if the data is null so that it runs only once. 
for eg.
```Java
if (speakerLiveData.getValue() != null && !reload)
            return speakerLiveData;
```
In this case instead of null check, I am just using a boolean flag.

**Comment by iamareebjamal** [comment]
OK

File: app/src/main/java/org/fossasia/openevent/app/core/main/EventViewModel.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 
 public class EventViewModel extends ViewModel {
     private final Preferences sharedPreferenceModel;
+    private boolean initialised = false;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/main/EventViewModel.java]: initialized

File: app/src/main/java/org/fossasia/openevent/app/core/main/EventViewModel.java [hunk 2]
```diff
@@ -43,6 +44,10 @@ public EventViewModel(Preferences sharedPreferenceModel, Bus bus,
     }
 
     protected void onStart() {
+        if (initialised) {
+            return;
+        }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/main/EventViewModel.java]: No curly brace for single statement

**Pull request merged**